emf4che build needs about 10Gb of memory

This is not a bug, but on my  linux environment (centos 7) the emf4che build fails unless I have about 10Gb of memory (physical plus swap).
The error message is not very clear, and it took me some time to find the cause, so maybe a note in the readme or requirements would help others if they hit the same problem (which is perhaps likely if compiling on a laptop). Increasing the swap size was the easy fix for me.
The issue seems to be the gwt-maven-plugin uses a lot of memory, so I don't think it is an issue with the emf4che build.
The error seen is:

[INFO] Che IDE :: Assembly :: Agent Server ................ SUCCESS [ 26.124 s]
[INFO] Che IDE :: Assembly :: Compiling GWT Application ... FAILURE [03:13 min]
[INFO] Che IDE :: Assembly :: Assemblies Tomcat ........... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 03:43 min (Wall Clock)
[INFO] Finished at: 2017-05-26T09:12:46+01:00
[INFO] Final Memory: 125M/1835M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:gwt-maven-plugin:2.8.0:compile (default) on project assembly-ide-war: Command [[
[ERROR] /bin/sh -c '/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.131-2.b11.el7_3.x86_64/jre/bin/java' '-Xss512m' '-Xmx2048m' 'com.google.gwt.dev.Compiler' '-logLevel' 'INFO' '-war' '/home/neil/git/emf4che/assembly/assembly-ide-war/target/assembly-ide-war-1.0.0-SNAPSHOT' '-localWorkers' '4' '-XfragmentCount' '-1' '-sourceLevel' '1.8' '-gen' '/home/neil/git/emf4che/assembly/assembly-ide-war/target/.generated' 'com.eclipsesource.emf.ide.IDE'
[ERROR] ]] failed with status 137

--- a/.gitignore
+++ b/.gitignore
@@ -59,6 +59,7 @@ Thumbs.db
 ######################
 .vagrant
 
-# Local testing data #
+# Local data/config #
 ######################
 data
+.mvn

--- a/.mvn/jvm.config
+++ b/.mvn/jvm.config
@@ -1,1 +1,1 @@
--Xmx4096m -Xms1024m -Djava.awt.headless=true -Djava.net.preferIPv4Stack=true -Dgwt.compiler.localWorkers=4
+-Xmx1024m -Xms512m -Djava.awt.headless=true -Djava.net.preferIPv4Stack=true -Dgwt.compiler.localWorkers=4

--- a/README.md
+++ b/README.md
@@ -37,6 +37,8 @@ mvn clean install
 
 For a faster build you can also run ```mvn -T 1C -DskipTests -Dskip-validate-sources clean install```
 
+Note: in case your build fails for no apparent reason you might have to reduce the memory settings of the JVM (used during the GWT build). Have a look at the .mvn/jvm.config file.
+
 # Run
 
 Next we want to start the customized Eclipse Che we built above. For this we need to use the [Eclipse Che CLI](https://www.eclipse.org/che/docs/setup/getting-started/index.html#volume-mounts). Use the following command and replace the `<>` tags with your local paths and version.

